File: Algorithm/C++/1000+/3612. Process String with Special Operations I.cc
#include <string>
#include <vector>
#include <cctype>
#include <algorithm>

class Solution {
public:
    std::string processStr(const std::string& s) {
        std::vector<char> result;
        for (char c : s) {
            if (std::isalpha(c)) {
                result.push_back(c);
            } else if (c == '*' && !result.empty()) {
                result.pop_back();
            } else if (c == '#') {
                result.insert(result.end(), result.begin(), result.end());
            } else if (c == '%') {
                std::reverse(result.begin(), result.end());
            }
        }
        return std::string(result.begin(), result.end());
    }
};

File: Algorithm/Python/1000+/3612. Process String with Special Operations I.py
class Solution:
    def processStr(self, s: str) -> str:
        result = []
        for c in s:
            if c.isalpha():
                result.append(c)
            elif c == '*' and len(result) > 0:
                result.pop()
            elif c == '#':
                result += result
            elif c == '%':
                result = result[::-1]
        return ''.join(result)
        